stress/common/src/main/java/alluxio/stress/worker/IOTaskSummary.java

```diff
@@ -0,0 +1,371 @@
+    mPoints = points;
+  }
+
+  /**
+   * @return the errors recorded
+   * */
+  public List<String> getErrors() {
+    return mErrors;
+  }
+
+  /**
+   * @param errors the errors
+   * */
+  public void setErrors(List<String> errors) {
+    mErrors = errors;
+  }
+
+  /**
+   * @return the {@link BaseParameters}
+   * */
+  public BaseParameters getBaseParameters() {
+    return mBaseParameters;
+  }
+
+  /**
+   * @param baseParameters the {@link BaseParameters}
+   * */
+  public void setBaseParameters(BaseParameters baseParameters) {
+    mBaseParameters = baseParameters;
+  }
+
+  /**
+   * @return the task specific {@link UfsIOParameters}
+   * */
+  public UfsIOParameters getParameters() {
+    return mParameters;
+  }
+
+  /**
+   * @param parameters the {@link UfsIOParameters}
+   * */
+  public void setParameters(UfsIOParameters parameters) {
+    mParameters = parameters;
+  }
+
+  /**
+   * @return the {@link SpeedStat} for read speed
+   * */
+  public SpeedStat getReadSpeedStat() {
+    return mReadSpeedStat;
+  }
+
+  /**
+   * @param stat the {@link SpeedStat} for read stats
+   * */
+  public void setReadSpeedStat(SpeedStat stat) {
+    mReadSpeedStat = stat;
+  }
+
+  /**
+   * @return the {@link SpeedStat} for write speed
+   * */
+  public SpeedStat getWriteSpeedStat() {
+    return mWriteSpeedStat;
+  }
+
+  /**
+   * @param stat the {@link SpeedStat} for write stats
+   * */
+  public void setWriteSpeedStat(SpeedStat stat) {
+    mWriteSpeedStat = stat;
+  }
+
+  /**
+   * An object representation of all the statistics we need
+   * from this I/O test.
+   * */
+  @JsonSerialize(using = StatSerializer.class)
+  @JsonDeserialize(using = StatDeserializer.class)
+  public static class SpeedStat implements JsonSerializable {
+    public double mTotalDuration; // in second
+    public long mTotalSize; // in Bytes
+    public double mMaxSpeed; // in MB/s
+    public double mMinSpeed; // in MB/s
+    public double mAvgSpeed; // in MB/s
+    public double mStdDev; // in MB/s
+
+    /**
+     * An empty constructor.
+     * */
+    public SpeedStat() {}
+
+    @Override
+    public String toString() {
+      return String.format("{totalDuration=%ss, totalSize=%s, maxSpeed=%sMB/s, "
+                      + "minSpeed=%sMB/s, " + "avgSpeed=%sMB/s, stdDev=%s}",
+              mTotalDuration, FormatUtils.getSizeFromBytes(mTotalSize),
+              mMaxSpeed, mMinSpeed, mAvgSpeed, mStdDev);
+    }
+  }
+
+  /**
+   * A customized serializer for {@link SpeedStat}.
+   * All the fields are attached with the unit.
+   * */
+  public static class StatSerializer extends StdSerializer<SpeedStat> {
+    private static final long serialVersionUID = -5198319303173120740L;
+
+    /**
+     * An empty constructor.
+     * */
+    public StatSerializer() {
+      super(SpeedStat.class);
+    }
+
+    @Override
+    public void serialize(SpeedStat value, JsonGenerator jgen, SerializerProvider provider)
```

Comment:
Why do serializers and deserializers need to be used? Jackson can just ser/de the classes, with getters/setters, or with public fields, right? That is how all the other results and parameter classes work.

Reply:
I provided customized ser/de because I need some better formatting on the field, so both the intermediate and final results are much more human readable.

Before formatting:
```
{"totalDuration":"2.0","totalSize":"415236096","maxSpeed":"200.0","minSpeed":"196.0","avgSpeed":"198.0","stdDev":2.8284271247461903}
```

After  formatting:
```
{"totalDuration":"2.0s","totalSize":"396.00MB","maxSpeed":"200.0MB/s","minSpeed":"196.0MB/s","avgSpeed":"198.0MB/s","stdDev":2.8284271247461903}
```

Reply:
I see, so is the json intended to be the final human-readable output? Typically, these json files are fed into generate, to get the human-readable output.

Reply:
Well this json will be read by jason's new UI, which will render a graph. So it doesn't need to be human readable

Reply:
Ah I see what you mean. Yes I intend to use this as the final output, as we are exploring the possibility to move the graph generation logic into out frontend. I think it also makes some sense if the user is reading the results before the GenerateReport step.

Reply:
Also I think it's slightly more maintainable if we pass data with units to the frontend, as it's a mixture of "Bytes", "MB", and "MB/s".

Reply:
I am a bit confused. If this json is the final human-readable output, then yes, these values can just be strings. But, if this json is supposed to be data (to be later graphed or displayed), then I think it would make sense for the value to be the scalar primitive. 

If units are a concern, other results have the units in the name of the field, like "durationMs", or "iompbs".

Reply:
Having units in the field name sounds like the way to go. I'll update this.